common/extension/rest_client.go

```diff
@@ -0,0 +1,20 @@
+package extension
+
+import (
+	"github.com/apache/dubbo-go/protocol/rest/rest_interface"
+)
+
+var (
+	restClients = make(map[string]func(restOptions *rest_interface.RestOptions) rest_interface.RestClient)
+)
+
+func SetRestClient(name string, fun func(restOptions *rest_interface.RestOptions) rest_interface.RestClient) {
+	restClients[name] = fun
+}
+
+func GetRestClient(name string, restOptions *rest_interface.RestOptions) rest_interface.RestClient {
+	if restClients[name] == nil {
+		panic("restClient for " + name + " is not existing, make sure you have import the package.")
+	}
+	return restClients[name](restOptions)
```

Comment:
why don't store instance in value? it will new a new object when you call ```GetRestClient```, will it necessary?

Reply:
Ok, I will store restClient using restOptions as a key of map.

Reply:
I think storing client in protocol is better.